parser/lexer.go

```diff
@@ -993,25 +1010,32 @@ func (r *reader) pos() Pos {
 }
 
 func (r *reader) updatePos(pos Pos) {
-	if r.p.Offset != pos.Offset {
-		r.peekRuneUpdated = false
-	}
 	r.p = pos
 }
 
 func (r *reader) data(from *Pos) string {
 	return r.s[from.Offset:r.p.Offset]
 }
 
-func (r *reader) incAsLongAs(fn func(rune) bool) rune {
+func (r *reader) incAsLongAs(fn func(b byte) bool) byte {
 	for {
 		ch := r.peek()
 		if !fn(ch) {
 			return ch
 		}
-		if ch == unicode.ReplacementChar && r.eof() {
+		if r.eof() {
 			return 0
 		}
 		r.inc()
 	}
 }
+
+// SkipRune skip mb character, return true indicate something has been skipped.
+func (r *reader) SkipRune() bool {
```

Comment:
How about using `Encoding` as the parameter? The function pointer `f` in `reader` is not needed anymore.

BTW, this function can be unexported.

Reply:
great